src/preFilter.ts

```diff
@@ -139,12 +141,40 @@ export function findBlockedStringInReleaseMaybe(
 	blockList: string[],
 ): string | undefined {
 	return blockList.find((blockedStr) => {
-		return (
-			searchee.title.includes(blockedStr) ||
-			(searchee.path &&
-				basename(dirname(searchee.path)).includes(blockedStr)) ||
-			blockedStr === searchee.infoHash
-		);
+		const { blocklistType, blocklistValue } =
+			parseBlocklistEntry(blockedStr)!;
+		switch (blocklistType) {
+			case BlocklistType.NAME:
+				return searchee.title.includes(blocklistValue);
+			case BlocklistType.NAME_REGEX:
+				return new RegExp(blocklistValue).test(searchee.title);
+			case BlocklistType.FOLDER:
+				return (
+					searchee.path &&
+					dirname(searchee.path).includes(blocklistValue)
+				);
+			case BlocklistType.FOLDER_REGEX:
+				return (
+					searchee.path &&
+					new RegExp(blocklistValue).test(dirname(searchee.path))
+				);
+			case BlocklistType.CATEGORY:
+				return blocklistValue === searchee.category;
+			case BlocklistType.TAG:
+				return searchee.tags?.includes(blocklistValue);
+			case BlocklistType.TRACKER:
+				return searchee.trackers?.some((tier) =>
+					tier.some((url) => url === blocklistValue),
+				);
+			case BlocklistType.HASH:
+				return blocklistValue === searchee.infoHash;
+			case BlocklistType.SIZE_BELOW:
+				return searchee.length < parseInt(blocklistValue);
+			case BlocklistType.SIZE_ABOVE:
+				return searchee.length > parseInt(blocklistValue);
+			default:
+				throw new Error(`Unknown blocklist type: ${blockedStr}`);
+		}
```

Comment:
I think we should try to make the new colon-separated blocklist entry format in a purely additive way—if there's no colon or the stuff to the left of a colon isn't one of the supported things, we can preserve logic to fall back to the legacy path/name/infohash matching and warn the user that it's deprecated.